File: Namrata-Sanger/66. Plus One.js
/*
Day 13
66. Plus One
Link: https://leetcode.com/problems/plus-one/
Level : Easy
*/

/**
 * @param {number[]} digits
 * @return {number[]}
 */

var plusOne = function(digits) {
    let carrier = 1;
    for(let i=digits.length - 1; i>=0; i--){
        let add = digits[i] + carrier;
        add = add.toString();
        if(add.length > 1){
                carrier = parseInt(add[0]);    
                digits[i] = parseInt(add[1]);
        }else{
            digits[i] = parseInt(add[0]);
            carrier = 0;
            break;
        }
    }
    if(carrier){
        digits = [carrier,...digits];
    }
    return digits;
};

File: Ayush-Gupta/66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int i, n = digits.size();
        for(i=n-1; i>=0; i--) {
            if(digits[i] == 9) digits[i] = 0;
            else {
                digits[i] += 1;
                return digits;
            }
        }
        digits.push_back(0);
        digits[0] = 1;
        return digits;
    }
};